app/components/doi-subject.js

```diff
@@ -1,77 +1,23 @@
 import Component from '@ember/component';
-import URI from 'urijs';
 import { isBlank } from '@ember/utils';
+import fosMapping from '../utils/fos-mappings';
 
-const completeSubjectList = [
-  'Natural sciences',
-  'Mathematics',
-  'Computer and information sciences',
-  'Physical sciences',
-  'Chemical sciences',
-  'Earth and related environmental sciences',
-  'Biological sciences',
-  'Other natural sciences',
-  'Engineering and technology',
-  'Civil engineering',
-  'Electrical engineering, electronic engineering, information engineering',
-  'Mechanical engineering',
-  'Chemical engineering',
-  'Materials engineering',
-  'Medical engineering',
-  'Environmental engineering',
-  'Environmental biotechnology',
-  'Industrial biotechnology',
-  'Nano-technology',
-  'Other engineering and technologies',
-  'Medical and health sciences',
-  'Basic medicine',
-  'Clinical medicine',
-  'Health sciences',
-  'Medical biotechnology',
-  'Other medical sciences',
-  'Agricultural sciences',
-  'Agriculture, forestry, and fisheries',
-  'Animal and dairy science',
-  'Veterinary science',
-  'Agricultural biotechnology',
-  'Other agricultural sciences',
-  'Social sciences',
-  'Psychology',
-  'Economics and business',
-  'Educational sciences',
-  'Sociology',
-  'Law',
-  'Political science',
-  'Social and economic geography',
-  'Media and communications',
-  'Other social sciences',
-  'Humanities',
-  'History and archaeology',
-  'Languages and literature',
-  'Philosophy, ethics and religion',
-  'Arts (arts, history of arts, performing arts, music)',
-  'Other humanities'
-];
-
-const oecdScheme = 'Fields of Science and Technology (FOS)';
-const oecdSchemeUri = 'http://www.oecd.org/science/inno/38235147.pdf';
+const completeSubjectList = fosMapping.allLabels();
 
 export default Component.extend({
+  attributeBindings: ['simple'],
   completeSubjectList,
   subjects: completeSubjectList,
   oecdSelected: false,
 
   init(...args) {
     this._super(...args);
-
     this.selected = this.selected || [];
   },
 
   setSchemeUri(value) {
-    let uri = URI(value);
     this.fragment.set('subjectSchemeUri', value);
-    this.fragment.set('valueUri', value);
```

Comment:
I was going to ask if there was a danger from removing the set here with respect to what the model sends back to the API (I had some issues with something similar with related item and null values), but I saw further down that you have set the default value to be `""` instead of `null`. Does this play nicely with the expectations of the main subjects control and what gets sent to the REST API? i.e if (and I haven't checked this so maybe it is a non-issue) in a DOI Subject the `valueUri` field is required, a blank string value is enough to satisfy that for Rails, as opposed to a `null` which will trigger an validation failure (this is the sort of area that it strikes me our existing testing might not cover so well, so it could slip by and accidentally cause an issue)

Reply:
The question is valid, especially considering the problems that have been caused in the past with a seemingly minor change like this, but after briefly checking it looks like there should be no problem with that.  Maybe adding a test would help?

Reply:
Yes the defult set to `""` instead of `null` is a conscious choice imposed by doing validation with json-schema.

In https://github.com/datacite/lupo/blob/3c9c99dfda93453591ec0a000ffbf0d2910e46f8/app/models/schemas/client/subjects.json#L10
`valueUri` must be a string.
 
In the API,  there is no validation for a DOI subject and it does not fail either way.  I think that in the future we should validate doi subjects, and there may be a fauture where we want to share the JSON schema to client applications so that feedback on JSON schema validations can be closer to the user.